src/meta/processors/jobMan/StatisJobExecutor.cpp

```diff
@@ -57,6 +57,56 @@ StatisJobExecutor::executeInternal(HostAddr&& address, std::vector<PartitionID>&
                                  std::move(parts), concurrency_, &(statisItem_[address]));
 }
 
+void showStatisItem(const cpp2::StatisItem& item, const std::string& msg) {
+    std::stringstream oss;
+    oss << msg << ": ";
+    oss << "tag_vertices: ";
+    for (auto& it : item.tag_vertices) {
+        oss << folly::sformat("[{}, {}] ", it.first, it.second);
+    }
+    oss << ", edges: ";
+    for (auto& it : item.edges) {
+        oss << folly::sformat("[{}, {}] ", it.first, it.second);
+    }
+    oss << folly::sformat(", space_vertices={}", item.space_vertices);
+    oss << folly::sformat(", space_edges={}", item.space_edges);
+    LOG(INFO) << oss.str();
+}
+
+void StatisJobExecutor::addStatis(cpp2::StatisItem& lhs, const cpp2::StatisItem& rhs) {
+    for (auto& it : rhs.tag_vertices) {
+        lhs.tag_vertices[it.first] += it.second;
+    }
+
+    for (auto& it : rhs.edges) {
+        lhs.edges[it.first] += it.second;
+    }
+
+    lhs.space_vertices += rhs.space_vertices;
+    lhs.space_edges += rhs.space_edges;
+    // ignore part_corelativity, they shoule be same
+}
+
+/**
+ * @brief caller will guarantee there won't be any conflict read / write.
+ */
+cpp2::ErrorCode StatisJobExecutor::saveSpecialTaskStatus(const cpp2::ReportTaskReq& req) {
+    if (!req.__isset.statis) {
+        return cpp2::ErrorCode::SUCCEEDED;
+    }
+    cpp2::StatisItem statisItem;
+    auto statisKey = MetaServiceUtils::statisKey(space_);
+    std::string val;
+    auto ret = kvstore_->get(kDefaultSpaceId, kDefaultPartId, statisKey, &val);
+    if (ret == kvstore::ResultCode::SUCCEEDED) {
```

Comment:
There maybe be a problem here:
If the last statis job and the current statis job are in the same space,
do you add the results of the last time?

Reply:
discussed offline, The previous status value will be init at prepare state.